watertap/unit_models/osmotically_assisted_reverse_osmosis_base.py

```diff
@@ -193,9 +195,18 @@ def build(self):
             doc="Isothermal assumption for permeate",
         )
         def eq_permeate_isothermal(b, t, x):
+            if x == self.length_domain.last():
+                return Constraint.Skip
             return (
-                b.feed_side.properties[t, x].temperature
-                == b.permeate_side.properties[t, x].temperature
+                b.permeate_side.properties[t, x].temperature
+                == 0.5
+                * b.feed_side.properties[
+                    t, b.feed_side.length_domain.first()
+                ].temperature
+                + 0.5
+                * b.permeate_side.properties[
+                    t, b.permeate_side.length_domain.last()
+                ].temperature
             )
```

Comment:
I'm not sure I understand your point -- this constraint as-expressed should work for the 1D model.

Reply:
I mean, for the 1D model. The temperature at permeate side apart from inlet is set as a same value (average of feed side and permeate side inlet). It might make more sense if we assign a value it node `i` as the average of `i-1` and `i+1`. I know it does not matter bacause we always have to set the same teperature otherwise we fail the initialization. But probably we could reconsider it later.

Reply:
Hmm maybe. This is an implementation of what @TimBartholomew suggested yesterday -- set all the nodes besides the inlet equal to the average.

I'm not sure it matters either way.